lib/code_corps/account/account.ex

```diff
@@ -0,0 +1,26 @@
+defmodule CodeCorps.Account do
+  @moduledoc ~S"""
+  Main entry-point for managing accounts.
+
+  All actions to acounts should go through here.
+  """
+
+  alias CodeCorps.{
+    GitHub.Adapters,
+    User,
+    Repo
+  }
+  alias Ecto.Changeset
+
+  @doc ~S"""
+  Creates a user record using attributes from a GitHub payload.
+  """
+  @spec create_from_github(map) :: {:ok, User.t} | {:error, Changeset.t}
+  def create_from_github(%{} = attrs) do
+    %User{}
+    |> Changeset.change(attrs |> Adapters.User.from_github_user())
+    |> Changeset.put_change(:context, "github")
+    |> Changeset.unique_constraint(:email)
```

Comment:
What happens if the email is `nil`? GitHub is not guaranteed to return one.

Reply:
We already merged in a PR where a user can be inserted without an email, so we should be fine. Really, what's important for us here is the github id, so we can associate this "stub" with an actual user later, when they connect with github.